src/utils/flow.js

```diff
@@ -36,14 +36,60 @@ export function immutableUpdate<T: Object>(object: T, ...rest: Object[]): T {
  * This function takes a string and returns either a valid TabSlug or null, this doesn't
  * throw an error so that any arbitrary string can be converted, e.g. from a URL.
  */
-export function toValidTabSlug(tabSlug: string): TabSlug | null {
-  switch (tabSlug) {
+export function toValidTabSlug(tabSlug: any): TabSlug | null {
+  const coercedTabSlug = (tabSlug: TabSlug);
+  switch (coercedTabSlug) {
     case 'calltree':
     case 'stack-chart':
     case 'marker-chart':
     case 'marker-table':
-      return tabSlug;
-    default:
+      return coercedTabSlug;
+    default: {
+      // The coerced type SHOULD be empty here. If in reality we get
+      // here, then it's not a valid transform type, so return null.
+      const unhandledCase: empty = coercedTabSlug; // eslint-disable-line no-unused-vars
```

Comment:
nit: you can just do `(coercedTabSlug: empty)` and get rid of the unused var.
(see https://flow.org/en/docs/types/casting/#toc-type-assertions)

Reply:
It's this eslint error or `no-unused-expressions` error

Reply:
Ah, I see. Then I'd suggest to go by the `no-unused-expressions` because it could be made to work with Flow assertions. I filed https://github.com/eslint/eslint/issues/9877 for this, maybe you can also reference the issue here.

Reply:
Adding it again so that it's embedded in a review:

Ah, I see. Then I'd suggest to go by the no-unused-expressions because it could be made to work with Flow assertions. I filed eslint/eslint#9877 for this, maybe you can also reference the issue here.